### PR TITLE
Kaggle Problem: Titanic - Basic ML Implementation

### PR DESCRIPTION
**Pull request by vibhorkrishna**
I have done this problem using various ML models and have used only basic knowledge of ML so that it is easy for beginners to understand. I have also explained the problem in readme file.
Please add hacktoberfest-accepted label to this pull request

### CODE REVIEW

**Comment by abhinand5** [comment]
Thanks for your contribution @vibhorkrishna 

I've reviewed your work, it looks good, but it would be even better if you can make these few changes...
1. Can you bring all that into a single jupyter notebook? (You can still keep the scripts if you want)
2. Can you write sections of your code in functions to avoid repeating the same chunk of code for multiple algorithms? 
3. Optional - EDA with output graphs

**Comment by vibhorkrishna** [comment]
I've made the following changes:

1. Brought all code into a single jupyter notebook
2. Written the code for multiple algorithms in functions
3. Placed some graphs. Was unable to place output graphs as I don't know the correct output of test dataset. Therefore was unable to make confusion matrix.

**Comment by abhinand5** [comment]
Great work @vibhorkrishna 
Once you do the following changes I'll merge it, thanks a lot for your contribution!

- [ ] You can just pull out `y_test` from test dataset as you have done for `X_test` and then you can use sklearn metrics module for `(y_test, y_pred)` to compute accuracy, confusion matrix and whatever you want.  
- [ ] Do all the import statements at the start of the notebook.

**Comment by vibhorkrishna** [comment]
There is no y_test in test dataset.
We only know the accuracy of the y_pred when we submit y_pred to kaggle submissions.
Otherwise I would have calculated confusion matrix.
I have written the accuracy of each model in README file by submitting my y_pred values one by one for each model in Kaggle.

**Comment by abhinand5** [comment]
Got it. Then,
- [ ] `train_test_split` from the train data itself and use that for validation. 
- [ ]  Do all the import statements at the start of the notebook.

**Comment by vibhorkrishna** [comment]
I've made the following changes:

1. Split the data using train_test split, trained the data and calculated the accuracy and confusion matrix
2. Made all the import statements at the start of the notebook

**Comment by vibhorkrishna** [comment]
As this issue is not in hacktoberfest. Therefore please label my pull request as hacktoberfest-accepted

**Comment by abhinand5** [comment]
Sure, Thanks for your contribution @vibhorkrishna

**Comment by abhinand5** [comment]
@vibhorkrishna Leave a star to help out this repo's visibility, if interested.

**Comment by vibhorkrishna** [comment]
Done

**Comment by RithikaVivaram** [comment]
By employing pipelines, I seamlessly integrated data preprocessing, feature engineering, model selection, and training into a unified and automated workflow, ensuring consistency and reproducibility in my machine-learning process.
